npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater/pnpm_lockfile_updater_spec.rb

```diff
@@ -144,6 +144,15 @@
       end
     end
 
+    context "when there is a private registry we don't have access to" do
+      let(:project_name) { "pnpm/private_package_access_with_package_name" }
+
+      it "raises a helpful error" do
+        expect { updated_pnpm_lock_content }
+          .to raise_error(Dependabot::PrivateSourceAuthenticationFailure)
```

Comment:
where is this new error thrown? I don't see the changes in this pR

Reply:
@abdulapopoola , function `raise_package_access_error` in  `npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/pnpm_lockfile_updater.rb:244` uses ` package_name = RegistryParser.new..` in `registry_parser.rb` to determine whether registry URL contains package name and raises either `PrivateSourceAuthenticationFailure` or `DependencyNotFound` (in case no @packagename was mentioned in url ) 

p.s : added more test coverage along with this at :156